src/test/compile-fail/invalid-module-declaration.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// ignore-tidy-linelength
+
+// error-pattern: cannot declare a new module at this location
+// note-pattern: maybe move this module `src/test/compile-fail/auxiliary/foo/bar` to its own directory via `src/test/compile-fail/auxiliary/foo/bar/mod.rs`
```

Comment:
`// note-pattern` has no meaning to the compile-fail tests -- use `// error-pattern` here instead (works just as well for notes).

Also, don't we want the pattern to be the following?
```
maybe move this module `src/test/compile-fail/auxiliary/foo/bar.rs` to its own directory via `src/test/compile-fail/auxiliary/foo/bar/mod.rs`
```